projects/components/src/description/description.component.ts

```diff
@@ -0,0 +1,39 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'ht-description',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  styleUrls: ['./description.component.scss'],
+  template: `
+    <div class="description">
+      <div
+        class="description-text"
+        [ngClass]="{ 'truncated-text': !this.isDescriptionTextToggled }"
+        data-sensitive-pii
+        #eventDescription
+      >
+        {{ description }}
+        <span *ngIf="this.isDescriptionTextToggled" (click)="this.toggleDescriptionText()" class="description-button"
+          >show less</span
+        >
+      </div>
+      <div
+        class="description-button"
+        *ngIf="eventDescription.offsetWidth < eventDescription.scrollWidth && !this.isDescriptionTextToggled"
```

Comment:
Also I found this ngIf not recalculate on  window resize, I tried to use ```@ViewChild``` decorator and several other ways but ref always return undefined, so it's another one issue I working on

Reply:
One option here is to use out layout change directive, which should give you a callback when the layout changes - either from a window resize, or a shift in other components. An example of this can be seen in `LabelComponent`